Coverage section of end2end test workflow needs to be skipped for Dependabot runs

### Describe the bug

From https://github.com/atsign-foundation/at_client_sdk/actions/runs/3597629173/jobs/6059571274

```
Run dart run coverage:format_coverage --check-ignore --lcov --in=coverage --out=end[2](https://github.com/atsign-foundation/at_client_sdk/actions/runs/3597629173/jobs/6059571274#step:9:2)end_test_coverage.lcov --report-on=lib
  dart run coverage:format_coverage --check-ignore --lcov --in=coverage --out=end2end_test_coverage.lcov --report-on=lib
  shell: /usr/bin/bash -e {0}
  env:
    PUB_CACHE: /home/runner/.pub-cache
    DART_HOME: /opt/hostedtoolcache/dart-sdk

Provided input "coverage" is neither a directory nor a file.

Usage: dart format_coverage.dart [OPTION...]

-s, --sdk-root               path to the SDK root
    --packages               [DEPRECATED] path to the package spec file
    --package                root directory of the package
                             (defaults to ".")
-i, --in                     input(s): may be file or directory
-o, --out                    output: may be file or stdout
                             (defaults to "stdout")
    --report-on              which directories or files to report coverage on
-j, --workers                number of workers
                             (defaults to "1")
    --bazel-workspace        Bazel workspace directory
                             (defaults to "")
-b, --base-directory         the base directory relative to which source paths are output
    --[no-]bazel             use Bazel-style path resolution
-r, --pretty-print           convert line coverage data to pretty print format
-f, --pretty-print-func      convert function coverage data to pretty print format
    --pretty-print-branch    convert branch coverage data to pretty print format
-l, --lcov                   convert coverage data to lcov format
-v, --verbose                verbose output
-c, --check-ignore           check for coverage ignore comments. Not supported in web coverage.
-h, --help                   show this help
Error: Process completed with exit code 1.
```

This block is failing because the previous block has been skipped.

### Steps to reproduce

1. Dependabot PR initiates CI run
2. Testing stage is skipped
3. No test results for coverage report
4. Coverage report barfs


### Expected behavior

Coverage report has the same skip rule as the test.

--- a/.github/workflows/at_client_sdk.yaml
+++ b/.github/workflows/at_client_sdk.yaml
@@ -183,10 +183,14 @@ jobs:
         run: dart test --concurrency=1 --coverage="coverage"
 
       - name: Convert coverage to LCOV format
+        # Needs tests to have run, so don't run for Dependabot PRs
+        if: ${{ github.actor != 'dependabot[bot]' }}
         working-directory: tests/at_end2end_test
         run: dart run coverage:format_coverage --check-ignore --lcov --in=coverage --out=end2end_test_coverage.lcov --report-on=lib
 
       - name: Upload coverage to Codecov
+        # Needs LCOV report to upload, so don't run for Dependabot PRs
+        if: ${{ github.actor != 'dependabot[bot]' }}
         uses: codecov/codecov-action@d9f34f8cd5cb3b3eb79b3e4b5dae3a16df499a70 # v3.1.1
         with:
           file: tests/at_end2end_test/end2end_test_coverage.lcov

